lib/init.coffee

```diff
@@ -11,4 +15,39 @@ module.exports =
         'Available codes: https://pypi.python.org/pypi/pep257#error-codes')
 
   activate: ->
-    console.log 'activate linter-pep257'
+    @subscriptions = new CompositeDisposable
+    @subscriptions.add atom.config.observe 'linter-pep257.executablePath',
+      (executablePath) =>
+        @executablePath = executablePath
+    @subscriptions.add atom.config.observe 'linter-pep257.ignoreCodes',
+      (ignoreCodes) =>
+        @ignoreCodes = ignoreCodes
+
+  deactivate: ->
+    @subscriptions.dispose()
+
+  lintPath: (filePath)->
+    params = [filePath]
+    if @ignoreCodes
+      params.push("--add-ignore=" + @ignoreCodes)
+    return helpers.exec(@executablePath, params, {stream: 'stderr'})
+
+  parseMessages: (output) ->
+    # parse lint output
+    output = output.replace(/:[\r\n]\s+/mg, " | ") # combine multiline output
+    messages = helpers.parse(output,
+                             "^(?<file>.+):(?<line>\\d+)\\s+(?<message>.+)")
+    messages.forEach (msg) ->
+      msg.type = "Info"
+    return messages
+
+  provideLinter: ->
+    provider =
+      grammarScopes: ['source.python']
+      scope: 'file'
+      lintOnFly: true
+      lint: (textEditor) =>
+        filePath = textEditor.getPath()
+        outputPromise = @lintPath(filePath)
+        messages = outputPromise.then(@parseMessages)
+        return messages
```

Comment:
You missed @steelbrain feedback before, to inline these guys as one.

I'm not entirely sure what the idiomatic coffee syntax is, but I got a feeling it's something like:
```coffee
return @lintPath filePath
  .then @parseMessages
```